lib/puppet/node/facts.rb

```diff
@@ -116,8 +116,25 @@ def add_timestamp
     @timestamp = Time.now
   end
 
+  def to_yaml
+    facts_to_display = Psych.parse_stream(YAML.dump(self))
+    quote_special_strings(facts_to_display)
+  end
+
   private
 
+  def quote_special_strings(fact_hash)
+    fact_hash.grep(Psych::Nodes::Scalar).each do |node|
+      next unless node.value =~ /:/
```

Comment:
Is this checking for values that start with `:`? If so, we'd need `/^:/ What's the significance of the color in a yaml scalar?

Reply:
The regex is for values that contains the character `:` anywhere (to quote ipv6 addresses also along with the macaddress)